WooCommerce/WooCommerceTests/ViewModels/MainTabViewModelTests.swift

```diff
@@ -0,0 +1,50 @@
+import XCTest
+
+@testable import WooCommerce
+
+import Yosemite
+
+/// Test cases for `MainTabViewModel`.
+final class MainTabViewModelTests: XCTestCase {
+
+    private var storesManager: MockupStoresManager!
+    private var featureFlagService: MockFeatureFlagService!
+
+    override func setUp() {
+        super.setUp()
+
+        var sessionManager = SessionManager.testingInstance
+        sessionManager.defaultCredentials = Credentials(authToken: "")
+
+        storesManager = MockupStoresManager(sessionManager: sessionManager)
+        featureFlagService = MockFeatureFlagService(isInAppFeedbackOn: true)
+    }
+
+    override func tearDown() {
+        featureFlagService = nil
+        storesManager = nil
+        super.tearDown()
+    }
+
+    func test_onViewDidAppear_will_save_the_installation_date() throws {
```

Comment:
I there a way to test this with the `DeauthenticatedState` state? 🤔

Reply:
Added in 30830df74bb390147a9974f4f849be46e8f186f5. 🤪